mobile/src/androidTest/java/ch/epfl/sdp/mobile/test/state/StatefulHomeTest.kt

```diff
@@ -177,4 +158,93 @@ class StatefulHomeTest {
 
     rule.onNodeWithContentDescription(strings.boardContentDescription).assertExists()
   }
+
+  @Test
+  fun creatingGameFromPrepareGameScreen_opensGameScreen() = runTest {
+    val auth = emptyAuth()
+    val store = buildStore {
+      collection("users") { document("userId2", ProfileDocument(name = "user2")) }
+    }
+    val authFacade = AuthenticationFacade(auth, store)
+    val social = SocialFacade(auth, store)
+    val chess = ChessFacade(auth, store)
+
+    authFacade.signUpWithEmail("user1@email", "user1", "password")
+    val currentUser = authFacade.currentUser.filterIsInstance<AuthenticatedUser>().first()
+    val user2 =
+        social.profile(uid = "userId2", user = currentUser).filterIsInstance<Profile>().first()
+    currentUser.follow(user2)
+
+    val strings =
+        rule.setContentWithLocalizedStrings {
+          ProvideFacades(authFacade, social, chess) { StatefulHome(currentUser) }
+        }
+
+    rule.onNodeWithText(strings.sectionPlay).performClick()
+    rule.onNodeWithText(strings.newGame).performClick()
+    rule.onNodeWithText("user2").performClick()
+    rule.onNodeWithText(strings.prepareGamePlay).performClick()
+    rule.onNodeWithContentDescription(strings.boardContentDescription).assertExists()
+  }
+
+  @Test
+  fun clickingOnPlayButtonFromPrepareGameScreen_withNoOpponentSelected_doesNothing() = runTest {
+    val auth = emptyAuth()
+    val store = buildStore {
+      collection("users") { document("userId2", ProfileDocument(name = "user2")) }
+    }
+    val authFacade = AuthenticationFacade(auth, store)
+    val social = SocialFacade(auth, store)
+    val chess = ChessFacade(auth, store)
+
+    authFacade.signUpWithEmail("user1@email", "user1", "password")
+    val currentUser = authFacade.currentUser.filterIsInstance<AuthenticatedUser>().first()
+    val user2 =
+        social.profile(uid = "userId2", user = currentUser).filterIsInstance<Profile>().first()
+    currentUser.follow(user2)
+
+    val strings =
+        rule.setContentWithLocalizedStrings {
+          ProvideFacades(authFacade, social, chess) { StatefulHome(currentUser) }
+        }
+
+    rule.onNodeWithText(strings.sectionPlay).performClick()
+    rule.onNodeWithText(strings.newGame).performClick()
+
+    rule.onNodeWithText(strings.prepareGamePlay).performClick()
+
+    rule.onNodeWithContentDescription(strings.boardContentDescription).assertDoesNotExist()
+    rule.onNodeWithText("user2").assertExists()
+  }
+
+  @Test
+  fun cancelingPreparegameScreen_returnsToPlaySection() = runTest {
+    val auth = emptyAuth()
+    val store = buildStore {
+      collection("users") { document("userId2", ProfileDocument(name = "user2")) }
+      collection("games") {}
```

Comment:
```suggestion
```

Reply:
Fixed in [9bcca9f](https://github.com/epfl-SDP/android/pull/200/commits/9bcca9f97a6022919e149265ea78cf9f4b472043)